community/values/src/main/java/org/neo4j/values/SequenceValue.java

```diff
@@ -52,4 +55,25 @@ default boolean equals( SequenceValue other )
     AnyValue value( int offset );
 
     int length();
+
+    default int compareToSequence( SequenceValue other, Comparator<AnyValue> comparator )
+    {
+        int i = 0;
+        int x = 0;
+        int length = Math.min( length(), other.length() );
+
+        while ( x == 0 && i < length )
+        {
+            x = comparator.compare( value( i ), other.value( i ) );
```

Comment:
I know this doesn't really change from how it used to work but not all ListValue implementations have constant time random access - `FilteredListValue` hasn't. So it would be more efficient - at least in that case - to use the `iterator` instead.

Reply:
Nice catch. Specialized both `compareTo` and `equals` now.